it/uk/gov/hmrc/fileupload/FileUploadRedirectionISpec.scala

```diff
@@ -0,0 +1,157 @@
+  val NOT_GOV_DOMAIN = "https://www.playframework.com/documentation/2.5.x/ScalaTestingWithScalaTest"
+
+  feature("Redirect End User when provided with redirect parameters as part of upload file request") {
+
+    val fileId = anyFileId
+    val envelopeId = anyEnvelopeId
+    scenario("Redirect upon success to valid url - only success url provided") {
+
+      Given("Envelope created with default parameters")
+      Wiremock.responseToUpload(envelopeId, fileId)
+      Wiremock.respondToEnvelopeCheck(envelopeId)
+
+      When("a file is uploaded provided a redirect on success to a https://www-dev.tax.service.gov.uk url")
+
+      val redirectSuccessUrl = DEV_YOUR_PAY
+      val queryParam = s"$SUCC_REDIRECT_PARAM_NAME=$redirectSuccessUrl"
+      val uploadFileResponse = uploadDummyFileWithoutRedirects(envelopeId, fileId, queryParam)
+      Then("upon success the user should be redirected to the url specified in the query parameter")
+      uploadFileResponse.status should be(301)
+      uploadFileResponse.header("Location").get shouldBe redirectSuccessUrl
+    }
+
+    scenario("Redirect upon success to valid url - both success and error urls provided") {
+
+      Given("Envelope created with default parameters")
+      Wiremock.responseToUpload(envelopeId, fileId)
+      Wiremock.respondToEnvelopeCheck(envelopeId)
+
+      When("a file is uploaded provided a redirect on success to a https://www-dev.tax.service.gov.uk url")
+      val redirectSuccessUrl = DEV_YOUR_PAY
+      val redirectErrorUrl = QA_YOUR_PAY
+
+      val queryParam = s"$SUCC_REDIRECT_PARAM_NAME=$redirectSuccessUrl&$FAIL_REDIRECT_PARAM_NAME=$redirectErrorUrl"
+      val uploadFileResponse = uploadDummyFileWithoutRedirects(envelopeId, fileId, queryParam)
+      Then("upon success the user should be redirected to the url specified in the query parameter")
+      uploadFileResponse.status should be(301)
+      uploadFileResponse.header("Location").get shouldBe redirectSuccessUrl
+    }
+
+    scenario("Redirect upon error to valid url - both success and error urls provided") {
+
+      Given("An envelope which does not exist")
+      val invalidEnvelopeId = EnvelopeId("12345-123124")
+
+      When("a file is uploaded provided a redirect on error to a https://www-dev.tax.service.gov.uk url")
+      val redirectSuccessUrl = "https://www-dev.tax.service.gov.uk/estimate-paye-take-home-pay/your-pay"
+      val redirectErrorUrl = "https://www-qa.tax.service.gov.uk/estimate-paye-take-home-pay/your-pay"
+      val queryParam = s"redirect-success-url=$redirectSuccessUrl&redirect-error-url=$redirectErrorUrl"
+      val uploadFileResponse = uploadDummyFileWithoutRedirects(invalidEnvelopeId, fileId, queryParam)
+
+      Then("upon success the user should be redirected to the url specified in the query parameter")
+      uploadFileResponse.status should be(301)
+      uploadFileResponse.header("Location").get.startsWith(redirectErrorUrl) shouldBe true
```

Comment:
"Some(...)" was not as string at Location header. It is just a test framework API since this header could not be provided.
Probably checking the returned parameters might have sense also.